src/hotspot/cpu/ppc/macroAssembler_ppc.cpp

```diff
@@ -1583,7 +1583,7 @@ void MacroAssembler::atomic_get_and_modify_generic(Register dest_current_value,
   // For older processors, instruction_type != size holds, and we
   // emulate the sub-word instructions by constructing a 4-byte value
   // that leaves the other bytes unchanged.
-  const int instruction_type = VM_Version::has_lqarx() ? size : 4;
+  const int instruction_type = size;
```

Comment:
I think it would be better to remove instruction_type and replace the remaining usages.